### DISCUSSION TITLE
Add RMA functionality

### DISCUSSION THREAD

**Discussion by lukeholder** [craftcms/commerce#2346] [Ideas]
Would fix #803 and #952

**Comment by engram-design** [comment]
To save people some Googling - **return merchandise authorisation (RMA)** .

In my scenario, one of our clients is pretty strict with their stock levels, and having cancelled orders should return stock to their products. I imagine this RMA system would handle that? My issue is similar to https://github.com/craftcms/commerce/issues/952

What's tricky - and I imagine this is what a larger system is for - is how to define when product stock needs to return. You might have two scenarios: 

1. Customer receives their product, but wants a refund (partial or otherwise). Stock shouldn't be added back, despite the refund.

2. Customer cancels their order after paid, but before shipped. Shop admin's change the order status to "Cancelled" and stock returns to the product. Their payment is refunded.

**Comment by yoannisj** [comment]
For consideration: I ran into a project which could use line-item refunds to create an internal document which details refunded items for accounting (much like an invoice with negative values).

Not sure what the RMA features would look like, but I guess if there was a way to extract the line items that were refunded based on the order's transactions or history, one could easily create an order document for this without guess-work.

**Comment by samhibberd** [comment]
@lukeholder about to embark on building out some custom line item refund logic, we really need it for a project we are about to launch. As it's fairly involved, I thought best check up on the status of this or similar stuff in development and whether it would impact our decision to write our own. Happy either way just be good to know the state of things?

**Comment by lenvanessen** [comment]
@samhibberd did you ever get around to building the logic yourself? Would love to see our approach, currently running into the same issue.

**Comment by yoannisj** [comment]
@lenvanessen , @samhibberd , I built a refund plugin which supports refunding and restocking individual line items as I needed this for a personal project. It have not added it to the plugin store, as I am still actively testing it but it just made production on that project that motivated building it:

[https://github.com/yoannisj/craft-order-refunds](https://github.com/yoannisj/craft-order-refunds)

I plan to release it very soon, but if you want to try it out on one of your projects let me know!

**Comment by jmauzyk** [comment]
Hi, I wanted to check if there are any plans around RMA / returns functionality in future releases of Commerce. We're looking to improve our returns process, and understanding whether this is something the team intends to leave to plugins or plans to address natively would help us determine our next steps. Thanks for any insights!

**Reply by lukeholder** [reply]
This is something on the list but we can't give any timeline. It would need be a custom module or plugin to provide this functionality. Thanks.
